pkg/nsx/services/vpc/vpc.go

```diff
@@ -549,21 +550,31 @@ func (s *VPCService) CreateorUpdateVPC(obj *v1alpha1.VPC) (*model.Vpc, *common.V
 	return &newVpc, &nc, nil
 }
 
-func (s *VPCService) Cleanup() error {
+func (s *VPCService) Cleanup(ctx context.Context) error {
 	vpcs := s.ListVPC()
 	log.Info("cleaning up vpcs", "Count", len(vpcs))
 	for _, vpc := range vpcs {
-		if err := s.DeleteVPC(*vpc.Path); err != nil {
-			return err
+		select {
+		case <-ctx.Done():
+			return util2.TimeoutFailed
```

Comment:
For all instances like this - can you actually at the very least wrap `ctx.Err()`? It will allow clients to check if the errors is [`context.DeadlineExceeded`](https://pkg.go.dev/context#pkg-variables).

Keeping your own error is fine too - but maybe something like:

```
return errors.Join(nsxutil.TimeoutFailed, ctx.Err())
```

?

Reply:
OK